mod/admin.php

```diff
@@ -209,9 +229,126 @@ function admin_content(&$a) {
 }
 
 /**
- * Admin Inspect Queue Page
+ * This function generates the "Federation Statistics" subpage for the admin
+ * panel. The page lists some numbers to the part of "The Federation" known to
+ * the node. This data includes the different connected networks (e.g.
+ * Diaspora, Hubzilla, GNU Social) and the used versions in the different
+ * networks.
+ *
+ * The returned string contains the HTML code of the subpage for display.
+ *
+ * @brief subpage with some stats about "the federation" network
+ * @param App $a
+ * @return string
+ */
+function admin_page_federation(&$a) {
+    // get counts on active friendica, diaspora, redmatrix, hubzilla, gnu
+    // social and statusnet nodes this node is knowing
+    //
+    // We are looking for the following platforms in the DB, "Red" should find
+    // all variants of that platform ID string as the q() function is stripping
+    // off one % two of them are needed in the query
+    // Add more platforms if you like, when one returns 0 known nodes it is not
+    // displayed on the stats page.
+    $platforms = array('Friendica', 'Diaspora', '%%red%%', 'Hubzilla', 'GNU Social', 'StatusNet');
+    $counts = array();
+    foreach ($platforms as $p) {
+	// get a total count for the platform, the name and version of the
+	// highest version and the protocol tpe
+	$c = q('select count(*), platform, network, version from gserver
+	    where platform like "'.$p.'" and last_contact > last_failure
+	    order by version asc;');
```

Comment:
It would be nicer if the query would formatted like this:

````
SELECT COUNT(*), `platform`, `network`, `version` FROM `gserver`
WHERE `platform` LIKE "'.$p.'" AND `last_contact` > `last_failure`
ORDER BY `version` ASC
````
And you could sanitize $p (although there is no need to do this in this case - but to stay consistent) and you may do it like always with "%s" placeholders.